R/dashboard.Rmd

```diff
@@ -1,28 +1,414 @@
+  
+  geom_tile(aes(y = Value_mill/2, height = Value_mill, width = 0.9),
+            alpha = 0.8, color = NA) +
+  
+  geom_text(aes(y = 0, label = paste(CTY_Name, " ")), vjust = 0.2, hjust = 1) +
+  geom_text(aes(y=Value_mill,label = Value_lbl, hjust=0)) +
+  
+  coord_flip(clip = "off", expand = FALSE) +
+  scale_y_continuous(labels = scales::comma) +
+  scale_x_reverse() +
+  guides(color = FALSE, fill = FALSE) +
+  
+  theme(axis.line=element_blank(),
+        axis.text.x=element_blank(),
+        axis.text.y=element_blank(),
+        axis.ticks=element_blank(),
+        axis.title.x=element_blank(),
+        axis.title.y=element_blank(),
+        legend.position="none",
+        panel.background=element_blank(),
+        panel.border=element_blank(),
+        panel.grid.major=element_blank(),
+        panel.grid.minor=element_blank(),
+        panel.grid.major.x = element_line( size=.1, color="grey" ),
+        panel.grid.minor.x = element_line( size=.1, color="grey" ),
+        plot.title=element_text(size=25, hjust=0.5, face="bold", colour="grey", vjust=-1),
+        plot.subtitle=element_text(size=18, hjust=0.5, face="italic", color="grey"),
+        plot.caption =element_text(size=8, hjust=0.5, face="italic", color="grey"),
+        plot.background=element_blank(),
+        plot.margin = margin(2,2, 2, 4, "cm")) +
+  
+  transition_states(Year, transition_length = 4, state_length = 1, wrap = FALSE) +
+  
+  view_follow(fixed_x = TRUE)  +
+  
+  labs(title = 'Export per Year : {closest_state}',  
+       subtitle  =  "Top 10 Countries Vanuatu has exported to",
+       caption  = "Export Value in Billions of Vatu | Data Source: Asycuda") 
+
+animate(anim_exp, 200)
+```
+
+
+```{r Export by commodities}
+startData <- toptenexport
+GBChoices <- as.list(names(startData))
+names(GBChoices) <- paste(names(startData),map(startData,~length(unique(.x))))
+
+updateData <- reactive(
+  startData %>% 
+    group_by_(input$GB) %>% 
+    summarise_if(is.numeric, sum, na.rm=T)
+  )
+```
+
+
+```{r Expot by countries}
+countryExport <- countryYearExport
+CEChoices <- as.list(names(countryExport))
+names(CEChoices) <- paste(names(countryExport),map(countryExport,~length(unique(.x))))
+
+updateCountryExport <- reactive(
+  countryExport %>%
+    group_by_(input$CE) %>% 
+    summarise_if(is.numeric,sum,na.rm=T)
+  )
+```
+
+
+Export by Commodity
+=====================================
+
+Column {.sidebar}
+------------------------------------------------------------------
+```{r}
+selectInput(inputId = "GB", label = "Group By", choices = GBChoices)
+selectInput(inputId = "Metric", label = "Metric", choices = names(select_if(startData,is.numeric)))
+```
+
+Column 
+------------------------------------------------------------------
+
+### Bar Plot of the selected commodity
+```{r}
+renderPlot({
+  updateData() %>%
+    ggplot(aes_string(x=input$GB, y=input$Metric, fill=input$GB)) +

```

Comment:
Using `aes_string()` function here - same as `aes()` function but takes column names as variables (or strings) to simplify the code